pkg/util/azureclient/storage/storage.go

```diff
@@ -93,3 +101,28 @@ type blob struct {
 }
 
 var _ Blob = &blob{}
+
+func (b *blob) CreateBlockBlobFromReader(blob io.Reader, options *storage.PutBlobOptions) error {
+	ctx := context.Background()
+	data, err := ioutil.ReadAll(blob)
+	if err != nil {
+		return err
+	}
+	sr := io.NewSectionReader(bytes.NewReader(data), 0, int64(len(data)))
+	return wait.PollImmediateUntil(2*time.Second,
```

Comment:
we are generally establishing a pattern why we try N times (3?) and log each failure.  Please do the same here.

Reply:
I remember why i didn't log, it's a long messy chain to pass the log variable into all the structs, but i can give it a go..